esmvalcore/preprocessor/_time.py

```diff
@@ -636,25 +637,32 @@ def anomalies(cube,
                                              seasons=seasons)
             cube = cube / cube_stddev
             cube.units = '1'
-        return cube
-
-    cube = _compute_anomalies(cube, reference, period, seasons)
-
-    # standardize the results if requested
-    if standardize:
-        cube_stddev = climate_statistics(cube,
-                                         operator='std_dev',
-                                         period=period)
-        tdim = cube.coord_dims('time')[0]
-        reps = cube.shape[tdim] / cube_stddev.shape[tdim]
-        if not reps % 1 == 0:
-            raise ValueError(
-                "Cannot safely apply preprocessor to this dataset, "
-                "since the full time period of this dataset is not "
-                f"a multiple of the period '{period}'")
-        cube.data = cube.core_data() / da.concatenate(
-            [cube_stddev.core_data() for _ in range(int(reps))], axis=tdim)
-        cube.units = '1'
+    else:
+        cube = _compute_anomalies(cube, reference, period, seasons)
+
+        # standardize the results if requested
+        if standardize:
+            cube_stddev = climate_statistics(cube,
+                                             operator='std_dev',
```

Comment:
eugh! we really need to deactivate in-line comments from codecov, it makes review really annoying :confounded:

Reply:
son of a moose, if you just press "a" on the keyboard you can toggle the codecove comments :grin: